core/unit_test/TestOccupancyControlTrait.hpp

```diff
@@ -0,0 +1,92 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include <Kokkos_Core.hpp>
+
+namespace {
+
+// Dummy policy for testing base class.
+template <class... Args>
+struct DummyPolicy : Kokkos::Impl::PolicyTraits<Args...> {
+  using execution_policy = DummyPolicy;
+};
+
+template <class... Properties>
+void test_policy_execution(const Kokkos::RangePolicy<Properties...>& policy) {
+  Kokkos::parallel_for(policy, KOKKOS_LAMBDA(int){});
+}
+template <class... Properties>
+void test_policy_execution(const Kokkos::TeamPolicy<Properties...>& policy) {
+  Kokkos::parallel_for(
+      policy,
+      KOKKOS_LAMBDA(
+          const typename Kokkos::TeamPolicy<Properties...>::member_type&){});
+}
+template <class... Properties>
+void test_policy_execution(const Kokkos::MDRangePolicy<Properties...>& policy) {
+  Kokkos::parallel_for(policy, KOKKOS_LAMBDA(int, int){});
+}
+template <class... Properties>
+void test_policy_execution(const DummyPolicy<Properties...>&) {}
```

Comment:
What is the point of having DummyPolicy in this test?

Reply:
Only to keep the runtime test I moved here.